File: quickbites-app/quickbites-backend/src/main/java/com/apsit/canteen_management/dto/SignupRequestDto.java
package com.apsit.canteen_management.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SignupRequestDto {
    private String username;
    private String password;
    private String mobileNumber;
    private String role;
}

File: quickbites-app/quickbites-user-app/canteen_user_app/lib/models/auth/signup_request_dto.dart
class SignupRequestDto {
  final String username;
  final String password;
  final String mobileNumber;
  final String role;

  const SignupRequestDto({
    required this.username,
    required this.password,
    required this.mobileNumber,
    required this.role,
  });

  factory SignupRequestDto.fromJson(Map<String, dynamic> json) {
    return SignupRequestDto(
      username: json['username'] as String,
      password: json['password'] as String,
      mobileNumber: json['mobileNumber'] as String,
      role: json['role'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'password': password,
      'mobileNumber': mobileNumber,
      'role': role,
    };
  }
}
